doc/book/routes.md

```diff
@@ -395,32 +396,57 @@ switch ($params['filter']) {
 }
 ```
 
+### Catch-all Parameters
+
+When a route may receive a variable number of parameters (for example, to
+implement a feature like echo, or to process an arbitrary list of files),
+you can use a catch-all parameter to collect all parameters that are not
+matched by another part of the route. These collected values can be accessed
+as a single parameter (whose name is defined in the route) containing an array.
+
+When used, the catch-all parameter must come after all positional value
+parameters. You can only use one catch-all parameter per route.
+
+Example:
+
+```
+say [loudly|softly]:volume [...words]
+```
+
+If the user entered the command line `say loudly I am here`, the 'volume'
+parameter would contain `'loudly'` and the 'words' parameter would contain
+`['I', 'am', 'here']`.
+
 ## Console routes cheat-sheet
 
 Param type                        | Example route definition    | Explanation
 :---                              | :---                        | :---
 **Literal params**                |                             |
 Literal                           | `foo bar`                   | "foo" followed by "bar"
-Literal alternative               | `foo (bar|baz)`             | "foo" followed by "bar" or "baz"
+Literal alternative               | `foo (bar\|baz)`             | "foo" followed by "bar" or "baz"
```

Comment:
Vertical bars within code tags don't need to be escaped. I'll fix that on merge, however.

Reply:
Strangely, the markdown was not rendering correctly on GitHub without the extra escapes -- I only added them because everything was garbled when I previewed my changes. Perhaps just a bug on the GitHub side, though!

Reply:
Not a bug. Different markdown libraries.

Reply:
Good to know. Thanks!